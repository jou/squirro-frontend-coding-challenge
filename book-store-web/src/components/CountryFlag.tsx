import { flagFromAlpha2Code } from "~/utils/EmojiFlag";
import { HTMLAttributes } from "react";

export interface CountryFlagProps {
    countryCode: string;
}

export default function CountryFlag({
    countryCode,
    ...spanProps
}: CountryFlagProps & HTMLAttributes<HTMLSpanElement>): JSX.Element {
    return <span {...spanProps}>{flagFromAlpha2Code(countryCode)}</span>;
}
