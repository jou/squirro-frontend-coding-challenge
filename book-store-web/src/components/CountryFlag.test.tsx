import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import CountryFlag from "~/components/CountryFlag";

describe("<CountryFlag>", () => {
    // Not much interesting happening here so far, a snapshot is enough for now
    it("should match snapshot", () => {
        const { container } = render(<CountryFlag countryCode="ch" />);

        expect(container).toMatchSnapshot();
    });
});
