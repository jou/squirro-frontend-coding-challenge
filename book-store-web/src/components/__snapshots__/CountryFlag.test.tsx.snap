// Vitest Snapshot v1

exports[`<CountryFlag> > should match snapshot 1`] = `
<div>
  <span>
    ğŸ‡¨ğŸ‡­
  </span>
</div>
`;
